fitsio/hdu/base.py

```diff
@@ -0,0 +1,351 @@
+            self._FITS.write_logical_key(self._ext+1,
+                                         str(name),
+                                         v,
+                                         str(comment))
+        elif isinstance(value, _stypes):
+            self._FITS.write_string_key(self._ext+1,
+                                        str(name),
+                                        str(value),
+                                        str(comment))
+        elif isinstance(value, _ftypes):
+            self._FITS.write_double_key(self._ext+1,
+                                        str(name),
+                                        float(value),
+                                        str(comment))
+        elif isinstance(value, _itypes):
+            self._FITS.write_long_key(self._ext+1,
+                                      str(name),
+                                      int(value),
+                                      str(comment))
+        elif isinstance(value, (tuple, list)):
+            vl = [str(el) for el in value]
+            sval = ','.join(vl)
+            self._FITS.write_string_key(self._ext+1,
+                                        str(name),
+                                        sval,
+                                        str(comment))
+        else:
+            sval = str(value)
+            mess = (
+                "warning, keyword '%s' has non-standard "
+                "value type %s, "
+                "Converting to string: '%s'")
+            warnings.warn(mess % (name, type(value), sval), FITSRuntimeWarning)
+            self._FITS.write_string_key(self._ext+1,
+                                        str(name),
+                                        sval,
+                                        str(comment))
+
+    def write_keys(self, records_in, clean=True):
+        """
+        Write the keywords to the header.
+
+        parameters
+        ----------
+        records: FITSHDR or list or dict
+            Can be one of these:
+                - FITSHDR object
+                - list of dictionaries containing 'name','value' and optionally
+                  a 'comment' field; the order is preserved.
+                - a dictionary of keyword-value pairs; no comments are written
+                  in this case, and the order is arbitrary.
+        clean: boolean
+            If True, trim out the standard fits header keywords that are
+            created on HDU creation, such as EXTEND, SIMPLE, STTYPE, TFORM,
+            TDIM, XTENSION, BITPIX, NAXIS, etc.
+
+        Notes
+        -----
+        Input keys named COMMENT and HISTORY are written using the
+        write_comment and write_history methods.
+        """
+
+        if isinstance(records_in, FITSHDR):
+            hdr = records_in
+        else:
+            hdr = FITSHDR(records_in)
+
+        if clean:
+            is_table = hasattr(self, '_table_type_str')
+            # is_table = isinstance(self, TableHDU)
```

Comment:
@esheldon This would cause a circular import in the new file structure. It also should probably be located in the TableHDU class and not here. Instead of doing that, I am simply looking for an attribute that is always set when a TableHDU class is made.